src/components/FieldValueDirectAnswer.tsx

```diff
@@ -0,0 +1,122 @@
+  BuiltInFieldType,
+  Address
+} from '@yext/search-headless-react';
+import { useMemo } from 'react';
+
+/**
+ * Props for FieldValueDirectAnswer.
+ */
+interface FieldValueDirectAnswerProps {
+  /** A field value direct answer result. */
+  result: FieldValueDirectAnswerType,
+  /** Handle onClick event for "View Details" link. */
+  viewDetailsClickHandler?: () => void,
+  /** CSS classes for customizing the component styling. */
+  cssClasses?: FieldValueDirectAnswerCssClasses
+}
+
+/**
+ *  The CSS class interface for FieldValueDirectAnswer.
+ */
+interface FieldValueDirectAnswerCssClasses {
+  header?: string,
+  body?: string,
+  content?: string,
+  answerContainer?: string
+}
+
+/**
+ * Renders a field value direct answer provided by the Search API.
+ */
+export function FieldValueDirectAnswer({
+  result,
+  viewDetailsClickHandler,
+  cssClasses = {}
+}: FieldValueDirectAnswerProps): JSX.Element {
+  const title = `${result.entityName} / ${result.fieldName}`;
+  const link = result.relatedResult.link ?? result.relatedResult.rawData.landingPageUrl as string;
+  const resultContent = useMemo(() => getResultContent(result), [result]);
+
+  return (
+    <div className={cssClasses.answerContainer}>
+      {title &&
+      <div className={cssClasses.header}>{title}</div>}
+      <div className={cssClasses.content}>
+        <div className={cssClasses.body}>{resultContent}</div>
+        {link && <div className='mt-4'>
+          <a href={link} className='text-primary' onClick={viewDetailsClickHandler}>
+            View Details
+          </a>
+        </div>
+        }
+      </div>
+    </div>
+  );
+}
+
+function getResultContent(result: FieldValueDirectAnswerType): JSX.Element {
+  switch (result.fieldType) {
+    case BuiltInFieldType.InstagramHandle:
+      return getUrlJsxElement(`https://www.instagram.com/${result.value}`, result.value);
+    case BuiltInFieldType.TwitterHandle:
+      return getUrlJsxElement(`https://twitter.com/${result.value}`, `@${result.value}`);
+    case BuiltInFieldType.FacebookURL:
+    case BuiltInFieldType.AndroidAppURL:
+    case BuiltInFieldType.IOSAppURL:
+      return getUrlJsxElement(result.value);
+    case BuiltInFieldType.ComplexURL:
+      const url = result.value.url;
+      const displayUrl = result.value.preferDisplayUrl ? result.value.displayUrl : url;
+      return getUrlJsxElement(url, displayUrl);
+    case BuiltInFieldType.URL:
+      return Array.isArray(result.value)
+        ? getListJsxElement(result.value, url => getUrlJsxElement(url))
+        : getUrlJsxElement(result.value);
+    case BuiltInFieldType.Phone:
+      return getUrlJsxElement(`tel:${result.value}`, result.value);
+    case BuiltInFieldType.Email:
+      return getListJsxElement(result.value, e => getUrlJsxElement(`mailto:${e}`, e));
+    case BuiltInFieldType.Address:
+      return getAddressJsxElement(result.value);
+    case BuiltInFieldType.RichText:
+      return <div>rich text</div>; //TODO: use react-markdown
```

Comment:
I guess we could still display the rich text raw string in the meantime? Do we have an item to use react-markdown?

Reply:
just made SLAP-2340 for that!

Reply:
looks like nidhi just made one so I can update the comment. sure I can display the raw string